### PR TITLE
Add ci-poseidon-e2e-gce for Poseidon project

### PR DESCRIPTION
**Pull request by shivramsrivastava**
This PR adds the E2E test job for Poseidon project.
I have verified it with kubetest, but with ```test-infra/jenkins/bootstrap.py``` i get this below error.
Is this because of some script error or some problem with my env?
This error happens on call to **'check_env'*** function.

```
W0426 14:41:14.484] NODE_NAMES=
I0426 14:41:14.584] Bringing down cluster
W0426 14:41:28.519] Deleted [https://www.googleapis.com/compute/v1/projects/poseidon-201815/zones/us-central1-b/disks/e2e-0-7de10-master-pd].
W0426 14:41:48.647] Deleted [https://www.googleapis.com/compute/v1/projects/poseidon-201815/global/firewalls/e2e-0-7de10-master-https].
W0426 14:41:49.984] Deleted [https://www.googleapis.com/compute/v1/projects/poseidon-201815/global/firewalls/e2e-0-7de10-master-etcd].
W0426 14:42:14.359] Deleted [https://www.googleapis.com/compute/v1/projects/poseidon-201815/global/firewalls/e2e-0-7de10-default-internal-master].
W0426 14:42:15.573] Deleted [https://www.googleapis.com/compute/v1/projects/poseidon-201815/global/firewalls/e2e-0-7de10-default-internal-node].
W0426 14:42:21.697] Deleted [https://www.googleapis.com/compute/v1/projects/poseidon-201815/global/firewalls/e2e-0-7de10-default-ssh].
I0426 14:42:23.220] Deleting firewall rules remaining in network e2e-0-7de10:
W0426 14:43:18.571] Deleted [https://www.googleapis.com/compute/v1/projects/poseidon-201815/global/networks/e2e-0-7de10].
I0426 14:43:18.978] Property "clusters.poseidon-201815_e2e-0-7de10" unset.
I0426 14:43:19.124] Property "users.poseidon-201815_e2e-0-7de10" unset.
I0426 14:43:19.262] Property "users.poseidon-201815_e2e-0-7de10-basic-auth" unset.
I0426 14:43:19.416] Property "contexts.poseidon-201815_e2e-0-7de10" unset.
I0426 14:43:19.420] Cleared config for poseidon-201815_e2e-0-7de10 from /tmp/whatever/github.com/kubernetes-sigs/poseidon/.kube/config
I0426 14:43:19.421] Done
W0426 14:43:19.433] 2018/04/26 14:43:19 process.go:152: Step './hack/e2e-internal/e2e-down.sh' finished in 2m9.429617194s
W0426 14:43:19.433] 2018/04/26 14:43:19 process.go:93: Saved XML output to /tmp/whatever/github.com/kubernetes-sigs/poseidon/_artifacts/junit_runner.xml.
W0426 14:43:19.434] 2018/04/26 14:43:19 main.go:309: Something went wrong: starting e2e cluster: error during ./hack/e2e-internal/e2e-up.sh: exit status 2
W0426 14:43:19.434] Traceback (most recent call last):
W0426 14:43:19.434]   File "/home/ubuntu/poseidongopathTest/src/k8s.io/test-infra/jenkins/../scenarios/kubernetes_e2e.py", line 729, in <module>
W0426 14:43:19.434]     main(parse_args())
W0426 14:43:19.434]   File "/home/ubuntu/poseidongopathTest/src/k8s.io/test-infra/jenkins/../scenarios/kubernetes_e2e.py", line 589, in main
W0426 14:43:19.434]     mode.start(runner_args)
W0426 14:43:19.435]   File "/home/ubuntu/poseidongopathTest/src/k8s.io/test-infra/jenkins/../scenarios/kubernetes_e2e.py", line 262, in start
W0426 14:43:19.435]     check_env(env, self.command, *args)
W0426 14:43:19.435]   File "/home/ubuntu/poseidongopathTest/src/k8s.io/test-infra/jenkins/../scenarios/kubernetes_e2e.py", line 111, in check_env
W0426 14:43:19.435]     subprocess.check_call(cmd, env=env)
W0426 14:43:19.435]   File "/usr/lib/python2.7/subprocess.py", line 541, in check_call
W0426 14:43:19.435]     raise CalledProcessError(retcode, cmd)
W0426 14:43:19.436] subprocess.CalledProcessError: Command '('kubetest', '--dump=/tmp/whatever/github.com/kubernetes-sigs/poseidon/_artifacts', '--gcp-service-account=/home/ubuntu/service-account.json', '--up', '--down', '--provider=gce', '--cluster=e2e-0-7de10', '--gcp-network=e2e-0-7de10', '--check-leaked-resources', '--extract=ci/latest', '--gcp-node-image=ubuntu', '--gcp-project=poseidon-201815', '--test-cmd=../test/e2e-poseidon-gce.sh', '--timeout=60m')' returned non-zero exit status 1
E0426 14:43:19.436] Command failed
I0426 14:43:19.436] process 29366 exited with code 1 after 9.0m
E0426 14:43:19.436] FAIL: ci-poseidon-e2e-gce
I0426 14:43:19.436] Call:  gcloud auth activate-service-account --key-file=/home/ubuntu/service-account.json
W0426 14:43:20.356] Activated service account credentials for: [poseidoncontainer@poseidon-201815.iam.gserviceaccount.com]
I0426 14:43:20.419] process 1141 exited with code 0 after 0.0m
I0426 14:43:20.420] Call:  gcloud config get-value account
I0426 14:43:21.032] process 1153 exited with code 0 after 0.0m
I0426 14:43:21.032] Will upload results to None using poseidoncontainer@poseidon-201815.iam.gserviceaccount.com

```

### CODE REVIEW

**Comment by shivramsrivastava** [comment]
@shashidharatd @deepak-vij @m1093782566

File: prow/config.yaml [hunk 1]
```diff
@@ -5175,6 +5175,24 @@ presubmits:
...
+      preset-k8s-ssh: true
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20180418-79a97ca02-master
+        args:
+        - "--job=$(JOB_NAME)"
+        - "--repo=github.com/$(REPO_OWNER)/$(REPO_NAME)=$(PULL_REFS)"
+        - "--root=/go/src"
+        - "--service-account=/etc/service-account/service-account.json"
+        - "--upload=gs://kubernetes-jenkins/pr-logs"
```
Comment (BenTheElder) [inline prow/config.yaml]: needs a timeout flag here

File: prow/config.yaml [hunk 2]
```diff
@@ -5175,6 +5175,24 @@ presubmits:
         - "--upload=gs://kubernetes-jenkins/pr-logs"
 
   kubernetes-sigs/poseidon:
+  - name: ci-poseidon-e2e-gce
+    interval: 24h
```
Comment (BenTheElder) [inline prow/config.yaml]: missing the `context` field for github

@cjwagner @stevekuznetsov do we really need separate "name" and "context" at this point ...?

Comment (BenTheElder) [reply prow/config.yaml]: (leveraging them for branched jobs could be nice, but IIRC we've been enforcing that they are identical for a long time now...)

**Comment by shivramsrivastava** [comment]
@BenTheElder added the timeout and context flag.

File: prow/config.yaml [hunk 3]
```diff
@@ -5175,6 +5175,26 @@ presubmits:
         - "--upload=gs://kubernetes-jenkins/pr-logs"
 
   kubernetes-sigs/poseidon:
+  - name: ci-poseidon-e2e-gce
+    context: ci-poseidon-e2e-gce
+    interval: 24h
+    always_run: true
```
Comment (BenTheElder) [inline prow/config.yaml]: er do you want an interval job or a postsubmit? for a postsubmit the interval field is meaningless. for a job that runs every 24 hours drop the `kubernetes-sigs/poseidon` bit, move it down below, and update the --repo flag to specify the repo without using the $env

**Comment by shivramsrivastava** [comment]
@BenTheElder I have removed the interval flag.
We want this job to run for every PR submit.

File: prow/config.yaml (hunk 2 above)
Comment (cjwagner) [reply prow/config.yaml]: I've been thinking that we should consider enforcing unique job names and support branch/RunIfChanged sharding by allowing jobs to specify the same context. I think that makes more sense intuitively and would could simplify the code in a number of places (having to search for the correct shard of a job based on the branch and changed files is cumbersome compared to looking up a job by its unique name).

There are some downsides to that though. I was hoping to get rid of the regexp triggers that jobs specify and instead require the job name to be used, but if we make different job shards have different names we would probably need to keep the regexp triggers around so that users don't have to worry about triggering the correct shard.

TLDR: As is, having both "name" and "context" isn't useful because they are always the same, but they do have semantic differences and I think it could be a good idea to keep them around because they may not always be the same if we change how sharding works.
